server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -714,25 +744,67 @@ private static ActionListener<SearchResponse> createCCSListener(
             countDown,
             skippedClusters,
             exceptions,
+            clusters.getCluster(clusterAlias),
             originalListener
         ) {
             @Override
             void innerOnResponse(SearchResponse searchResponse) {
+                // TODO: in CCS fail fast ticket we may need to fail the query if the cluster is marked as FAILED
+                ccsClusterInfoUpdate(searchResponse, cluster, skipUnavailable);
+                Instant now = Instant.now();
+                Instant start = Instant.ofEpochMilli(startTime);
+                cluster.setTook(Duration.between(start, now).toMillis());
```

Comment:
can you remind me why we need to compute the took time here? is it not included in the search response obtained from the remote cluster?